src/metabase/lib/drill_thru/common.cljc

```diff
@@ -72,30 +72,41 @@
-  ;; If a breakout-sourced column comes from a model based on a native query that renames the column with an "AS"
-  ;; alias, AND where the column has been mapped to a real DB field, then we can't use the breakout column directly
-  ;; and must instead lookup the equivalent "resolved" column metadata. This results in a (hopefully) equivalent
-  ;; column where the :lib/source is no longer :source/breakouts, but rather :source/card, which allows
-  ;; column-metadata->field-ref to recognize that it needs to generate a named-based ref. This is required because an
-  ;; id-based ref will wind up generating SQL that matches the underlying mapped column's name, not the name of the
-  ;; column from the model's native query (which was renamed via "AS").
-  ;;
-  ;; https://github.com/metabase/metabase/issues/53556
-  ;; https://metaboat.slack.com/archives/C0645JP1W81/p1739904084459979
-  (if-not (possible-model-mapped-breakout-column? query column)
-    column
-    (let [resolved-column  (lib.metadata.calculation/metadata query stage-number (lib.ref/ref column))
-          underlying-unit  (::lib.underlying/temporal-unit column)
-          matching-column  (some-> resolved-column
-                                   (m/assoc-some ::lib.underlying/temporal-unit underlying-unit))]
-      (or matching-column column))))
+  ([query stage-number column] (breakout->resolved-column query stage-number column nil))
+  ([query        :- ::lib.schema/query
+    stage-number :- :int
+    column       :- ::lib.schema.metadata/column
+    {:keys [preserve-bucketing?]}]
+   ;; TODO: This is a hack to workaround field refs confusion that should be fixed by the field refs overhaul. Remove
+   ;; this function and possible-model-mapped-breakout-column?, above, once the field refs overhaul lands.
+   ;;
+   ;; If a breakout-sourced column comes from a model based on a native query that renames the column with an "AS"
+   ;; alias, AND where the column has been mapped to a real DB field, then we can't use the breakout column directly
+   ;; and must instead lookup the equivalent "resolved" column metadata. This results in a (hopefully) equivalent
+   ;; column where the :lib/source is no longer :source/breakouts, but rather :source/card, which allows
+   ;; column-metadata->field-ref to recognize that it needs to generate a named-based ref. This is required because an
+   ;; id-based ref will wind up generating SQL that matches the underlying mapped column's name, not the name of the
+   ;; column from the model's native query (which was renamed via "AS").
+   ;;
+   ;; https://github.com/metabase/metabase/issues/53556
+   ;; https://metaboat.slack.com/archives/C0645JP1W81/p1739904084459979
+   (if-not (or (possible-model-mapped-breakout-column? query column)
+               (day-bucketed-breakout-column? column))
+     column
+     (let [field-ref (cond-> (lib.ref/ref column)
+                       (not preserve-bucketing?) (update 1 dissoc :base-type :effective-type))
```

Comment:
Maybe I'm confused about terminology, but I was expecting `preserve-bucketing?` to preserve (or not) the temporal-unit rather than the types.

Reply:
Good catch. 👍